server/server.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+)
+
+// New returns an unstarted HTTP server with health and proxy handlers.
+func New(logger *log.Logger, config *api.Config, token string) *http.Server {
+	mux := http.ServeMux{}
+	mux.HandleFunc("/", proxyHandler(logger, config, token))
+	srv := http.Server{
+		Handler: &mux,
+	}
+
+	return &srv
+}
+
+// The proxyHandler borrows from the Send function in Vault Agent's proxy:
+// https://github.com/hashicorp/vault/blob/22b486b651b8956d32fb24e77cef4050df7094b6/command/agent/cache/api_proxy.go
+func proxyHandler(logger *log.Logger, config *api.Config, token string) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		logger.Printf("Proxying %s %s\n", r.Method, r.URL.Path)
+		fwReq, err := proxyRequest(r, config.Address, token)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("failed to generate proxy request: %s", err), http.StatusInternalServerError)
+			return
+		}
+
+		resp, err := config.HttpClient.Do(fwReq)
```

Comment:
Any reason not to use retryablehttp.Client here?

Reply:
I don't have strong opinions here, but I erred on the side of simplicity in order to delegate more complex behaviour like retrying and error handling to the proxy's client. The less the proxy does, the more transparent it is to the client who can mostly act as though it's talking directly to Vault. Do you have a preference to add retries?

Reply:
After syncing offline, we decided to leave this as-is for the first version. We may add retries in a future revision, as long as we can be confident about what requests are safe to retry (idempotent etc).